src/main/gov/nasa/jpf/psyco/search/datastructures/searchImage/StateImage.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2015, United States Government, as represented by the 
+ * Administrator of the National Aeronautics and Space Administration.
+ * All rights reserved.
+ *
+ * The PSYCO: A Predicate-based Symbolic Compositional Reasoning environment 
+ * platform is licensed under the Apache License, Version 2.0 (the "License"); you 
+ * may not use this file except in compliance with the License. You may obtain a 
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0. 
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed 
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR 
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the 
+ * specific language governing permissions and limitations under the License.
+ */
+package gov.nasa.jpf.psyco.search.datastructures.searchImage;
+
+import gov.nasa.jpf.psyco.search.datastructures.region.Region;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A state image is the current search state at a given point of time.
+ * At the end of the search, an StateImage object is returned as search result.
+ * @param <T> Any type implementing the abstract Region class.
+ */
+public abstract class StateImage<T extends Region> {
```

Comment:
StateImage is a strange name for this class. Is this the image of some state under some function?